manager/flags.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manager
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+)
+
+type Uri struct {
```

Comment:
I don't think this is good for the flag interface, yeah. Gets too much done too soon.

I didn't do the REST API part yet, what I did for now is simply change the underlying type to be stronger without changing any of the behaviour.

I seem to recall you said that you'd like the endpoint to use something other than the strings that the flags use. Would that be for GET, POST or both? In a way I'd rather keep it the way it is now, since it's more intuitive and it uses the same format everywhere.

Although at the same time, a json detailed description of each sink would be useful. Perhaps a separate endpoint, something like `/api/v1/sink-configs`?